lib/galaxy/util/checkers.py

```diff
@@ -49,10 +50,27 @@ def check_binary(name, file_path=True):
     # Handles files if file_path is True or text if file_path is False
     if file_path:
         temp = open(name, "rb")
+        file_size = os.stat(name).st_size
+        read_start = int(file_size / 2)
+        # avoid reading beyond the end of the file
+        read_length = min(1024, int(file_size / 2) - 1)
     else:
         temp = BytesIO(name)
+        length = len(name)
+        read_start = int(length / 2)
+        read_length = min(1024, int(length / 2) - 1)
```

Comment:
Hi @pcm32 
This is a small refactoring suggestion:
```suggestion
    if file_path:
        temp = open(name, "rb")
        size = os.stat(name).st_size
    else:
        temp = BytesIO(name)
        size = len(name)
    read_start = int(size / 2)
    read_length = 1024
```

If I'm not mistaken, the `read([bytes])` function won't read beyond EOF so it's safe to fix the `read_length` to the value you wish to read at most.

Reply:
Sounds good.